daemon/logger/jsonfilelog/read.go

```diff
@@ -174,6 +175,15 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 		fileWatcher.Close()
 	}()
 
+	logWatcherClosed := make(chan struct{})
+	go func() {
```

Comment:
Looking at the watch loop below, I think we need some way to cancel this goroutine since the loop may exit without logwatcher closing. May be time to whip out a context.